### ISSUE TITLE
[Questions] There are no unit tests in the project

### ISSUE THREAD

**Issue by suho** [y4gamilight/nimble-assignment#53]
## Issue

Currently, there are no unit tests in the project. It's a requirement of the code challenge, could you share with me why you did not add any tests?

**Comment by y4gamilight** [comment]
Yeah, I received the document and need at least a day to confirm with the team about the version. So I couldn't start early, I also requested a deadline of 14 instead of 12, but it was rejected. 
I accepted that I missed one of the requirements, and submitted the app without enough features. 
On the other hand, the code base follows DI  to implement unit tests in the next phase
